modules/auxiliary/scanner/rsync/modules_list.rb

```diff
@@ -25,147 +23,92 @@ def initialize
 
-  def rsync_list
-    sock.puts("#list\n")
+    sock.get # server_motd
 
-    list = []
-    # the module listing is the module name and comment separated by a tab, each module
-    # on its own line, lines separated with a newline
-    sock.get(read_timeout).split(/\n/).map(&:strip).map do |module_line|
-      next if module_line =~ /^#{RSYNC_HEADER} EXIT$/
-      name, comment = module_line.split(/\t/).map(&:strip)
-      list << [ name, comment ]
-    end
+    sock.puts(version) # client_initialisation
+    sock.puts(dir) # client_query
 
-    list
+    data = sock.get # module_list
+    data.gsub!('@RSYNCD: EXIT', '') if data # Final module list
+    disconnect
+    data
   end
 
-  # Attempts to negotiate the rsync protocol with the endpoint.
-  def rsync_negotiate(get_motd)
-    # rsync is promiscuous and will send the negotitation and motd
-    # upon connecting.  abort if we get nothing
-    return unless greeting = sock.get_once
-
-    # parse the greeting control and data lines.  With some systems, the data
-    # lines at this point will be the motd.
-    greeting_control_lines, greeting_data_lines = rsync_parse_lines(greeting)
-
-    # locate the rsync negotiation and complete it by just echo'ing
-    # back the same rsync version that it sent us
-    version = nil
-    greeting_control_lines.map do |greeting_control_line|
-      if /^#{RSYNC_HEADER} (?<version>\d+(\.\d+)?)$/ =~ greeting_control_line
-        version = Regexp.last_match('version')
-        sock.puts("#{RSYNC_HEADER} #{version}\n")
-      end
+  def auth?(dir)
+    data = rsync(dir)
+    if data && data =~ /RSYNCD: OK/m
+      false
+    else
+      true
     end
+  end
 
-    unless version
-      vprint_error("#{ip}:#{rport} - no rsync negotation found")
-      return
-    end
+  def module_list_format(module_list)
+    mods = []
+    rows = []
 
-    motd_lines = greeting_data_lines
-    if get_motd
-      _, post_neg_data_lines = rsync_parse_lines(sock.get_once)
-      motd_lines |= post_neg_data_lines
-    end
-    [ version, motd_lines.empty? ? nil : motd_lines.join("\n") ]
-  end
+    return if module_list.blank?
 
-  # parses the control and data lines from the provided response data
-  def rsync_parse_lines(response_data)
-    control_lines = []
-    data_lines = []
-
-    if response_data
-      response_data.strip!
-      response_data.split(/\n/).map do |line|
-        if line =~ /^#{RSYNC_HEADER}/
-          control_lines << line
-        else
-          data_lines << line
-        end
+    module_list = module_list.strip
+    module_list = module_list.split("\n")
+
+    module_list.each do |mod|
+      name, desc = mod.split("\t")
+      name = name.strip
+      mods << { name: name, desc: desc } if name
+
+      if datastore['AUTH_CHECK']
+        rows << [name, desc, "#{auth?(name)}"]
+      else
+        rows << [name, desc, 'Unknown']
       end
     end
 
-    [ control_lines, data_lines ]
+    table = Msf::Ui::Console::Table.new(
```

Comment:
If `THREADS` > 1, this will be printed out but it won't be clear which host it belongs to.